lib/ansible/plugins/inventory/ini.py

```diff
@@ -106,8 +123,21 @@ def __init__(self) -> None:
         self._origin: Origin | None = None
 
     def verify_file(self, path):
-        # hardcode exclusion for TOML to prevent partial parsing of things we know we don't want
-        return super().verify_file(path) and os.path.splitext(path)[1] != '.toml'
+
+        valid = super().verify_file(path)
+
+        p = Path(path)
+        allowed = self.get_option('allowed_extensions')
+        if valid:
+            if allowed:
+                if p.suffix[1:] not in allowed:
+                    valid = False
+            elif p.suffix[1:] != 'ini':
+                self.display.deprecated(
+                    'Usage of ini inventory plugin with an inventory source that does not have the "ini" extension or the "allowed_extensions" option is not set',
```

Comment:
Suggest a more terse deprecation description with `help_text` for the prescriptive part, e.g.:
```
self.display.deprecated(f"Parsed inventory with invalid extension {p.suffix!r}.", help_text="Change the file extension to '.ini' or configure 'allowed_extensions'.", version="2.23")
```